web/packages/selfhosted/test/utils.ts

```diff
@@ -200,3 +200,28 @@ export function loadJsAPI(swf?: string) {
         }
     });
 }
+
+export async function closeAllModals(
+    browser: WebdriverIO.Browser,
+    player: WebdriverIO.Element,
+) {
+    await browser.execute(
+        (modals) => {
+            for (const modal of modals) {
+                const m = modal as unknown as HTMLElement;
+                const cl = m.querySelector(".close-modal")! as HTMLElement;
+                cl.click();
+            }
+        },
+        await player.shadow$$(".modal:not(.hidden)"),
```

Comment:
webdriverio v9 automatically pierces the shadow DOM.

Reply:
Should be good now!